babel/dates.py

```diff
@@ -994,6 +1161,25 @@ def __getitem__(self, name):
         else:
             raise KeyError('Unsupported date/time field %r' % char)
 
+    def extract(self, char):
+        char = str(char)[0]
+        if char == 'y':
+            return self.value.year
+        elif char == 'M':
+            return self.value.month
+        elif char == 'd':
+            return self.value.day
+        elif char == 'H':
+            return self.value.hour
+        elif char == 'h':
+            return (self.value.hour % 12 or 12)
```

Comment:
It would be kind of neat to see a test method specifically for `extract`, but no big deal, feel free to ignore.